### DISCUSSION TITLE
ignore auxvariables in exodiff?

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#15821] [Q&A General] [answered]
Hi -

Is it possible to have the testharness or exodiff only look at differences in nonlinear variables and ignore auxvariables or even ignore the fact that auxvariables might be missing that are in the gold file? Ideally I'd like to still have the auxvariables be in the file for analysis.

**Comment by aeslaughter** [comment, marked as answer]
Yes, this is possible using a custom comparison file which can be generated. The file can be created with exodiff and specified in the tests specification with the 'custom_cmp' parameter. I don't know how to make that file, but perhaps @bwspenc @dschwen could help.

I usually prefer to just remove the variables from the output, there is a 'hide' option that you can use within the Outputs block.

```
[Outputs]
  [out]
    type = Exodus
    hide = 'aux0 aux1'
  []
[]
```
